### ISSUE TITLE
load might support a cache mechanism

### ISSUE THREAD

**Issue by femtotrader** [queryverse/Queryverse.jl#6]
Hello,

http://www.david-anthoff.com/jl4ds/stable/fileio.html shows usage such as

    df = load("https://raw.githubusercontent.com/davidanthoff/CSVFiles.jl/master/test/data.csv") |> DataFrame

maybe a cache mechanism (such as https://github.com/helgee/RemoteFiles.jl or https://github.com/oxinabox/DataDeps.jl ) should be integrated to Queryverse (or at least its usage be better documented)

A possible API could be

    df = load(RemoteFile("https://raw.githubusercontent.com/davidanthoff/CSVFiles.jl/master/test/data.csv", "data.csv")) |> DataFrame

or

    df = load(RemoteFile("https://raw.githubusercontent.com/davidanthoff/CSVFiles.jl/master/test/data.csv")) |> DataFrame

So file will be stored with `data.csv` as filename (by default) and inside a default directory

Pinging @oxinabox and @helgee

But this is probably not the right place to suggest this... maybe FileIO?

Kind regards

**Comment by oxinabox** [comment]
Using DataDeps with this should be trivial. 

something like (not tested so maybe has typos)

```
using DataDeps

function __init__()
    register(DataDep("Queryverse Tests", 
        "Testing data for queryverse", 
        "https://raw.githubusercontent.com/davidanthoff/CSVFiles.jl/master/test/data.csv"
    ))
end

df = load(datadep"Queryverse Tests/data.csv") |> DataFrame
```

In fact,
one of the [examples on my blog](https://white.ucc.asn.au/2018/05/28/Optimizing-your-diet-with-JuMP.html)

is using DataDeps with ExcelFiles.jl which is part of queryverse right?

**Comment by oxinabox** [comment]
I guess I am missing why this wants to actually be part of queryverse, or FileIO.jl

Queryverses in built functionality to work directly on URL is just a convenience for when it is convenient (small data, or single use scripts).
Beyond that the user can just use other packages for external cached fetching,
and such packages are going to address a wider range of surrounding issues than trying to capture this functionality as an extension to the existing convenience functionality.

To be clear,
Is it just a docs thing?
 to improve discoverability across the Julia ecosystem?

**Comment by femtotrader** [comment]
Yes I think for DataDeps a doc effort is required into Queryverse to give it some highlight.

**Comment by femtotrader** [comment]
But I think some parameter of `DataDep` (such as Message) should be optional

**Comment by femtotrader** [comment]
With [RemoteFiles.jl](https://github.com/helgee/RemoteFiles.jl), it can be achieved using

```julia
using Queryverse
using RemoteFiles
@RemoteFile(DATA, "https://raw.githubusercontent.com/davidanthoff/CSVFiles.jl/master/test/data.csv")
download(DATA)
df = load(path(DATA)) |> DataFrame
```

but maybe a better API should be defined.
